src/components/Clock.tsx

```diff
@@ -0,0 +1,42 @@
+import { Component } from 'react';
+
+type Props = {
+  name: string,
+};
+
+type State = {
+  date: Date,
+};
+
+export class Clock extends Component<Props, State> {
+  state: State = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      let { date } = this.state;
+
+      date = new Date();
```

Comment:
You don't need state.date here:
```suggestion
      const date = new Date();
```